### PR TITLE
adls filesystem implementation

### PR DESCRIPTION
**Pull request by dwaijam**

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.IMRU.Tests/app.config [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+<configuration>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-10.0.0.0" newVersion="10.0.0.0" />
+      </dependentAssembly>
+    </assemblyBinding>
+  </runtime>
+</configuration>
```
Comment (homezcx) [inline lang/cs/Org.Apache.REEF.IMRU.Tests/app.config]: new line required.

File: lang/cs/Org.Apache.REEF.All/app.config [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+<configuration>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-10.0.0.0" newVersion="10.0.0.0" />
+      </dependentAssembly>
+    </assemblyBinding>
+  </runtime>
+</configuration>
```
Comment (homezcx) [inline lang/cs/Org.Apache.REEF.All/app.config]: You probably want to run check style to fix format. here new line required. And several other files.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
+﻿using System;
```
Comment (homezcx) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs]: missing license. And several other files.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+        [Inject]
+        private AzureDataLakeFileSystem(AzureDataLakeStoreClient client)
+        {
+            _client = client.GetReference();
+        }
+
+        public void Copy(Uri sourceUri, Uri destinationUri)
+        {
+            throw new NotImplementedException();
```
Comment (homezcx) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs]: If it is not intermediate code, I think a message is needed in this exception "Copy() is not implemented for AzureDataLakeFileSystem."

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        }
+
+        public void CreateDirectory(Uri directoryUri)
+        {
+            _client.CreateDirectory(directoryUri.ToString());
+        }
+
+        public Uri CreateUriForPath(string path)
+        {
+            throw new NotImplementedException();
```
Comment (homezcx) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs]: same here and line 75

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        }
+
+        public bool Exists(Uri fileUri)
+        {
+            return _client.CheckExists(fileUri.ToString());
+        }
+
+        public IEnumerable<Uri> GetChildren(Uri directoryUri)
+        {
+            foreach (var entry in _client.EnumerateDirectory("/Test"))
```
Comment (homezcx) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs]: Curious: what is "/Test" here?

File: lang/cs/Org.Apache.REEF.Bridge.CLR/app.config [hunk 7]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.Bridge.CLR/app.config]: Why are all these Newtonsoft.Json assemblies being added?

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureDataLakeFileSystem.cs [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+using Org.Apache.REEF.IO.FileSystem;
+using Org.Apache.REEF.IO.FileSystem.AzureDataLake;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Xunit;
+
+namespace Org.Apache.REEF.IO.Tests
+{
+    public sealed class TestAzureDataLakeFileSystem
+    {
+        private readonly static Uri FakeUri = new Uri("http://fake.com");
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO.Tests/TestAzureDataLakeFileSystem.cs]: Even though it's fake data, the URI should match the expected format of an ADLS URI.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        }
+
+        public void Copy(Uri sourceUri, Uri destinationUri)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void CopyFromLocal(string localFileName, Uri remoteFileUri)
+        {
+            _client.BulkUpload(localFileName, remoteFileUri.ToString());
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs]: BulkUpload is an asynchronous method but the IFileSystem interface has these methods as synchronous. You will need to implement the wait. Same goes for all the asynchronous AdlsClient methods you are using.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+        }
+
+        public Uri CreateUriForPath(string path)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void Delete(Uri fileUri)
+        {
+            _client.Delete(fileUri.ToString());
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs]: Although the interface has a void function, the ADLSClient returns a Boolean for delete. You need to reconcile this difference. I suggest throwing an IOException if the result is false. The same goes for the other ADLSClient boolean functions.

**Pull request closed without merging**